### PR TITLE
src: kw_env_show: show the current env in terminal.

### PR DESCRIPTION
**Pull request by aquilamacedo**
This commit adds support for automatically
showing the current environment in the terminal.

Closes: #762

### CODE REVIEW

**Comment by aquilamacedo** [comment]
Hi @rodrigosiqueira. 

I opened this draft because some things are left to finish, like tests, comments, and giving an overview of the code. I'm still going to change a lot of things. Do you have any recommendations regarding the structure of the code?

File: kw [hunk 1]
```diff
@@ -255,6 +255,13 @@ function kw()
         env_main "$@"
       )
       ;;
+    env-show)
+      (
+        include "${KW_LIB_DIR}/kw_env_show.sh"
+
+        env_show_main "$@"
+      )
+      ;;
```
Comment (rodrigosiqueira) [inline kw:264]: This should be configured during the setup time. Maybe you can create a new option for the `setup.sh` script.

File: src/kw_env_show.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    install_home
+    return "$?"
+  fi
+}
+
+function install_home()
+{
+  say 'Installing ...'
+
+  if [[ -f "$HOME/.bashrc" ]]; then
```
Comment (rodrigosiqueira) [inline src/kw_env_show.sh:21]: String concatenation. Use kw code style.

File: src/kw_env_show.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    update_path ".bashrc"
+  else
+    warning ".bashrc file not found."
```
Comment (rodrigosiqueira) [inline src/kw_env_show.sh:24]: Prefer single quotes if you do not need to expand a variable.

Comment (rodrigosiqueira) [reply src/kw_env_show.sh:24]: How about ZSH users?

File: src/kw_env_show.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+{
+  local shellrc="$1"
+
+  local content
+
+  content=$(
+    cat << 'EOF'
+parse_git_file() { git_file=$(git rev-parse --show-toplevel 2>/dev/null)/.kw/env.current; [ -f "$git_file" ] && printf '[%s]' "$(cat "$git_file")"; }
+
+PS1='\[\e[0;34m\]\u@\h \[\e[0;32m\]\w\[\e[0;36m\] $(parse_git_file)\[\e[0m\] $ '
```
Comment (rodrigosiqueira) [inline src/kw_env_show.sh:38]: Maybe we can create a file with this code in the etc/ folder and load it when we need it.

File: src/kw_env_show.sh [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+EOF
+  )
+
+  bashrc_append "${content}" "${HOME}/${shellrc}"
+}
+
+function bashrc_append()
+{
+  # TODO: create a conditional that checks if this
+  # argument code $1 already exists inside the .bashrc
```
Comment (rodrigosiqueira) [inline src/kw_env_show.sh:48]: Don't forget to address this TODO.

File: src/kw_env_show.sh [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    esac
+  done
+}
+
+function env_show_help()
+{
+  printf '%s\n' 'kw env-show:' \
+    '  env [-i | --install] - List all environments available' \
+    '  env (-d | --help) - Destroy an environment'
+}
```
Comment (rodrigosiqueira) [inline src/kw_env_show.sh:94]: We don't need this.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

**Comment by rodrigosiqueira** [comment]
@aquilamacedo  please, update this PR otherwise, I'll close it.

**Comment by aquilamacedo** [comment]
> @aquilamacedo please, update this PR; otherwise, I'll close it.

I'm still working on the changes, I'll try to update as soon as possible

File: etc/strings/current_env_content.txt
Comment (rodrigosiqueira) [inline etc/strings/current_env_content.txt:1]: Why you saved this file in the strings folder? The string folder is dedicated to text to be loaded when we need to print some message to the users.

File: setup.sh [hunk 7]
```diff
@@ -448,6 +449,26 @@ function install_home()
...
 
+function show_current_env()
+{
+  local config_files_path="$etcdir"
+  local config_file_template="$config_files_path/strings/current_env_content.txt"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ -f "$config_file_template" ]]; then
+    content_current_env_file=$(cat ${config_file_template})
```
Comment (rodrigosiqueira) [inline setup.sh]: See: https://kworkflow.org/content/codingstyle.html#command-substitution-and-arithmetic-expression

File: setup.sh [hunk 8]
```diff
@@ -448,6 +449,26 @@ function install_home()
   update_version
 }
 
+function show_current_env()
+{
+  local config_files_path="$etcdir"
+  local config_file_template="$config_files_path/strings/current_env_content.txt"
```
Comment (rodrigosiqueira) [inline setup.sh:455]: https://kworkflow.org/content/codingstyle.html#string-concatenation

File: setup.sh [hunk 9]
```diff
@@ -448,6 +449,26 @@ function install_home()
...
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ -f "$config_file_template" ]]; then
+    content_current_env_file="$(cat "${config_file_template}")"
+    if ! grep -Fxq "$content_current_env_file" "$shellrc"; then
```
Comment (rodrigosiqueira) [inline setup.sh:461]: Run this command outside this `if` and use the full parameter name, e.g., use --fixed-strings instead of -F.

File: setup.sh [hunk 10]
```diff
@@ -448,6 +449,26 @@ function install_home()
...
 
+function show_current_env()
+{
+  local config_files_path="$etcdir"
+  local config_file_template="$config_files_path/strings/current_env_content.txt"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ -f "$config_file_template" ]]; then
+    content_current_env_file="$(cat "${config_file_template}")"
```
Comment (rodrigosiqueira) [inline setup.sh:460]: https://kworkflow.org/content/codingstyle.html#command-substitution-and-arithmetic-expression

File: setup.sh [hunk 11]
```diff
@@ -448,6 +449,26 @@ function install_home()
...
+    if ! grep -Fxq "$content_current_env_file" "$shellrc"; then
+      safe_append "$content_current_env_file" "$shellrc"
+      warning ' -> For a better experience with kw, please, open a new terminal.'
+    fi
+  else
+    warning "Setup could not find ${config_file_template}"
+    warning 'Try running ./setup -i'
+    return 2
```
Comment (rodrigosiqueira) [inline setup.sh:468]: Add the 2 error comment.

File: etc/strings/current_env_content.txt [hunk 12]
```diff
@@ -0,0 +1,2 @@
+parse_git_file() { git_file=$(git rev-parse --show-toplevel 2>/dev/null)/.kw/env.current; [ -f "$git_file" ] && printf '[%s]' "$(cat "$git_file")"; }
```
Comment (rodrigosiqueira) [inline etc/strings/current_env_content.txt:1]: The string folder is only for strings that will be displayed to users. You could keep this under the src folder.

Additionally, why `parse_git_file`? Is this a copy & paste issue?

File: setup.sh [hunk 13]
```diff
@@ -196,6 +196,7 @@ function usage()
   say '--help      | -h     Display this usage message'
   say "--install   | -i     Install $app_name"
   say "--uninstall | -u     Uninstall $app_name"
+  say "--show-env  | -s     Show current env in terminal"
```
Comment (rodrigosiqueira) [inline setup.sh:199]: Maybe `--setup-env-notification`.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

**Comment by aquilamacedo** [comment]
I made some changes and created the `setup.sh --show-env` option.

But I still have a question, would implementing this code inside the `setup.sh --install` option be better? Something like a question during installation: **"Do you want to customize the terminal to show the current env?"** 

Or is the `--show-env` option itself enough? What do you think?

This GIF shows an example of how it's getting:

2023-03-17-20-24-29-_online-video-cutter com_

File: etc/strings/current_env_content.txt
Comment (aquilamacedo) [reply etc/strings/current_env_content.txt:1]: oh, sorry I didn't know, I'll fix it

File: etc/strings/current_env_content.txt (hunk 12 above)
Comment (aquilamacedo) [reply etc/strings/current_env_content.txt:1]: I forgot to change that name; it's not intuitive at all. I had left this name while testing how to make it show the contents of _env.current_ in the terminal.

**Comment by rodrigosiqueira** [comment]
@aquilamacedo ,
When I try your PR, I have this error:

```
Setup could not find /home/siqueira/.config/kw/strings/current_env_content.txt
Try running ./setup -i
```

About your question, we can add this to the `setup -i`, but we should ask the user if they want that. However, we should also provide this option in the setup.

Please, carefully update this PR by addressing all of my comments and test: remove env, add env, switch env, and exit env.

**Comment by aquilamacedo** [comment]
> @aquilamacedo , When I try your PR, I have this error:
> 
> ```
> Setup could not find /home/siqueira/.config/kw/strings/current_env_content.txt
> Try running ./setup -i
> ```
> 
> About your question, we can add this to the `setup -i`, but we should ask the user if they want that. However, we should also provide this option in the setup.
> 
> Please, carefully update this PR by addressing all of my comments and test: remove env, add env, switch env, and exit env.

Ok, I'm already working on it.

File: etc/strings/current_env_content.txt (hunk 12 above)
Comment (aquilamacedo) [reply etc/strings/current_env_content.txt:1]: Wouldn't it be better to save this file in /etc instead of /src?

File: setup.sh (hunk 11 above)
Comment (aquilamacedo) [reply setup.sh:468]: I didn't understand very well. Would it be writing a comment explaining why to run `./setup -i`?

**Comment by aquilamacedo** [comment]
I made the requested changes, but I still need to fix a bug I saw in **_.bashrc_**; it seems that when running `./setup.sh -i` more than once gives a problem because it appears that some other code is rewriting the code of **_etc/current_env_content.txt_**; I'm still fixing this bug.

The GIF below is an example of testing it: removing env, switching env and exiting env.

2023-03-19-00-17-06-_online-video-cutter com_

File: setup.sh [hunk 14]
```diff
@@ -193,14 +193,15 @@ function usage()
...
-  say "--install   | -i     Install $app_name"
-  say "--uninstall | -u     Uninstall $app_name"
-  say '--skip-checks        Skip checks (use this when packaging)'
-  say '--verbose            Explain what is being done'
-  say '--force              Never prompt'
-  say "--completely-remove  Remove $app_name and all files under its responsibility"
-  say "--docs               Build $app_name's documentation as HTML pages into ./build"
+  say '--help                    | -h     Display this usage message'
+  say "--install                 | -i     Install $app_name"
+  say "--uninstall               | -u     Uninstall $app_name"
```
Comment (rodrigosiqueira) [inline setup.sh:198]: These changes and others in this help should be in its own commit.

File: setup.sh [hunk 15]
```diff
@@ -193,14 +193,15 @@ function usage()
...
+  say "--setup-env-notification  | -s     Show current env in terminal"
+  say '--skip-checks             Skip checks (use this when packaging)'
+  say '--verbose                 Explain what is being done'
+  say '--force                   Never prompt'
+  say "--completely-remove       Remove $app_name and all files under its responsibility"
```
Comment (rodrigosiqueira) [inline setup.sh:203]: When creating the new commit that changes this part, use the string concatenation style defined by kw.

File: setup.sh [hunk 16]
```diff
@@ -446,6 +447,30 @@ function install_home()
   remove_legacy_git_from_kernel_config_manager
   # Update version based on the current branch
   update_version
+  # Show current environment in terminal
+  show_current_env
+}
+
+function show_current_env()
```
Comment (rodrigosiqueira) [inline setup.sh:454]: I still don't like this function name because this is more like a setup than an action like 'show'... please, find a better name for that... maybe look for some ideas from the git project.

File: setup.sh [hunk 17]
```diff
@@ -446,6 +447,30 @@ function install_home()
...
+{
+  local config_files_path="$etcdir"
+  local config_file_template="${config_files_path}/current_env_content.txt"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ $(ask_yN "Do you want to customize the bash terminal to show the current env?") =~ '1' ]]; then
```
Comment (rodrigosiqueira) [inline setup.sh:461]: Single quotes around the `ask_yN` parameter.

File: setup.sh [hunk 18]
```diff
@@ -446,6 +447,30 @@ function install_home()
...
+    if [[ -f "$config_file_template" ]]; then
+      content_current_env_file="$(< "$config_file_template")"
+    else
+      warning "Setup could not find ${config_file_template}"
+      return 2
+    fi
+    if ! grep --fixed-strings --line-regexp --quiet "$content_current_env_file" "$shellrc"; then
```
Comment (rodrigosiqueira) [inline setup.sh:468]: Please, move this grep outside this `if` and just check `$?`.

File: setup.sh [hunk 19]
```diff
@@ -446,6 +447,30 @@ function install_home()
...
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ $(ask_yN "Do you want to customize the bash terminal to show the current env?") =~ '1' ]]; then
+    if [[ -f "$config_file_template" ]]; then
+      content_current_env_file="$(< "$config_file_template")"
+    else
+      warning "Setup could not find ${config_file_template}"
+      return 2
+    fi
```
Comment (rodrigosiqueira) [inline setup.sh:467]: Add a new line after this `fi`.

File: setup.sh [hunk 20]
```diff
@@ -446,6 +447,30 @@ function install_home()
...
+  local config_file_template="${config_files_path}/current_env_content.txt"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ $(ask_yN "Do you want to customize the bash terminal to show the current env?") =~ '1' ]]; then
+    if [[ -f "$config_file_template" ]]; then
+      content_current_env_file="$(< "$config_file_template")"
+    else
+      warning "Setup could not find ${config_file_template}"
+      return 2
```
Comment (rodrigosiqueira) [inline setup.sh:466]: Please, always document the return error. See kw code for more examples.

File: etc/current_env_content.txt [hunk 21]
```diff
@@ -0,0 +1,2 @@
+kw_show_current_env() { kw_env_file=$(git rev-parse --show-toplevel 2>/dev/null)/.kw/env.current; [ -f "$kw_env_file" ] && printf '[%s]' "$(cat "$kw_env_file")"; }
```
Comment (rodrigosiqueira) [inline etc/current_env_content.txt:1]: Why this code need to looks like this? Can you put it as a standard function in a readable way? Also, use the `.sh` extension or no extension. Finally, this file name does not describe this script well... how git names something like this?

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi,
Please figure out what's is wrong with the new version and address all my comments in this PR

File: setup.sh (hunk 16 above)
Comment (aquilamacedo) [reply setup.sh:454]: How about _setup_bashrc_current_env_custom()_ ?

File: setup.sh (hunk 19 above)
Comment (aquilamacedo) [reply setup.sh:467]: Ok, I'll add this line.

File: setup.sh (hunk 20 above)
Comment (aquilamacedo) [reply setup.sh:466]: Nice, I'll  document.

File: etc/current_env_content.txt (hunk 21 above)
Comment (aquilamacedo) [reply etc/current_env_content.txt:1]: I had compacted this code into two lines so as not to take up too much space in the code, but now moving it to etc makes it easier to extend the code again. As for the name, I'll think of something better.

File: setup.sh (hunk 14 above)
Comment (aquilamacedo) [reply setup.sh:198]: I didn't understand very well, you say create a new commit just for this change?

File: setup.sh (hunk 15 above)
Comment (aquilamacedo) [reply setup.sh:203]: nice!!

File: etc/current_env_content.txt (hunk 21 above)
Comment (aquilamacedo) [reply etc/current_env_content.txt:1]: Using the readable/extended code, it isn't very easy to check if this code already exists inside the _.bashrc_. Could you give me a recommendation?

image

Comment (rodrigosiqueira) [reply etc/current_env_content.txt:1]: I don't think you should inject `kw_get_current_env_name()` in the bashrc this way. Keep this code in a file under kw folder, and just include it in the bashrc.

Comment (rodrigosiqueira) [reply etc/current_env_content.txt:1]: Also, you should not use `PS1=`; this will override any previous PS1 configuration. It would be best if you first ensure that the `kw_get_current_env_name` is available; after that, you need should use PS1+=.

File: setup.sh (hunk 14 above)
Comment (rodrigosiqueira) [reply setup.sh:198]: These changes are not part of this commit, and they have a different context. For this reason, they should have their commit. The only expectation is the `--setup-env-notification`.

File: setup.sh (hunk 16 above)
Comment (rodrigosiqueira) [reply setup.sh:454]: Maybe `setup_bashrc_to_show_current_kw_env`

File: etc/git_prompt_current_env_name.sh [hunk 22]
```diff
@@ -0,0 +1,7 @@
+kw_get_current_env_name()
+{
+  kw_env_file=$(git rev-parse --show-toplevel 2> /dev/null)/.kw/env.current
```
Comment (rodrigosiqueira) [inline etc/git_prompt_current_env_name.sh:3]: I don't think we need `git rev-parse`. Why not use `$PWD`? Keep in mind that kw might work in a kernel without git.

File: setup.sh [hunk 23]
```diff
@@ -446,6 +447,32 @@ function install_home()
...
+}
+
+function setup_bashrc_to_show_current_kw_env()
+{
+  local config_files_path="$etcdir"
+  local config_file_template="${config_files_path}/git_prompt_current_env_name.sh"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ $(ask_yN 'Do you want to customize the bash terminal to show the current env?') =~ '1' ]]; then
```
Comment (rodrigosiqueira) [inline setup.sh:461]: If the user selects `--setup-env-notification` it means they want this feature. In other words, it does not make sense to ask this question. You may wish to alert that the bashrc will be updated or something like that.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Is this going to work on zsh?

**Comment by rodrigosiqueira** [comment]
@aquilamacedo any update here? This PR has been open for a long time...

File: etc/git_prompt_current_env_name.sh [hunk 24]
```diff
@@ -0,0 +1,7 @@
+kw_get_current_env_name()
```
[the remark this answers is not in the document]
Comment (aquilamacedo) [reply etc/git_prompt_current_env_name.sh:1]: ok, i'll change this.

**Comment by aquilamacedo** [comment]
> @aquilamacedo any update here? This PR has been open for a long time...

I'm almost done, I found a small bug when updating the current env in the terminal. I'm almost finishing the PR.

**Comment by aquilamacedo** [comment]
> Is this going to work on zsh?

No. This commit only supported bash. For ZSH I believe the implementation will be a little different and will need a dedicated commit just for ZSH.

**Comment by rodrigosiqueira** [comment]
After testing your change and thinking about it more carefully, we should change the behavior slightly.

# Run `./setup -i`

We should not touch the `.bashrc` file. However, at the end of the installation, we should give this info:

```
Info:
kw provides integration to the terminal by telling users about the current env in use. If you want to use it, add the following line to your .bashrc:

PS1+=(YOUR COMMAND HERE)
```

# Run `./setup -s`

We should not provide the modification option; we just need to provide info similar to the one described in the previous topic.

# Expand documentation and tutorial

Expand the documentation to explain how to add the env info in the bashrc.

File: documentation/tutorials/kw_environment_prompt.rst [hunk 25]
```diff
@@ -0,0 +1,50 @@
+=======================================================================
```
Comment (rodrigosiqueira) [inline documentation/tutorials/kw_environment_prompt.rst:1]: You don't need to create a new tutorial; just expand the `kw env` tutorial.

https://kworkflow.org/tutorials/kw-env.html

File: documentation/tutorials/kw_environment_prompt.rst [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+
+.. note::
+    If you have modified the PS1 variable in the .bashrc file, you will need to adapt
+    it to work correctly. You can simply call the ``kw_get_current_env_name`` function
+    on PS1 to display the current environment being used in kworkflow.
+
+Conclusion
+----------
+
+In summary, customizing your bash terminal to display the current environment in kworkflow
```
Comment (rodrigosiqueira) [inline documentation/tutorials/kw_environment_prompt.rst:49]: You will not need this part after you merge this file with the kw env tutorial.

File: documentation/tutorials/setup.rst [hunk 27]
```diff
@@ -40,6 +40,10 @@ Then install **kw**::
 
   ./setup.sh --install
 
+If you want to costumize your terminal to see the environment::
```
Comment (rodrigosiqueira) [inline documentation/tutorials/setup.rst:43]: Add a link to the env tutorial.

File: setup.sh [hunk 28]
```diff
@@ -196,6 +196,7 @@ function usage()
   say '--help      | -h     Display this usage message'
   say "--install   | -i     Install $app_name"
   say "--uninstall | -u     Uninstall $app_name"
+  say "--setup-kw-env-prompt | -s  Show current env in terminal"
```
Comment (rodrigosiqueira) [inline setup.sh:199]: Use sigle-quotes.

File: setup.sh [hunk 29]
```diff
@@ -401,6 +401,11 @@ function safe_append()
   fi
 }
 
+function print_separator()
```
Comment (rodrigosiqueira) [inline setup.sh:404]: Do we really need a function for this?

File: setup.sh [hunk 30]
```diff
@@ -446,6 +451,35 @@ function install_home()
...
+{
+  local config_files_path="$etcdir"
+  local config_file_template="${config_files_path}/kw_prompt_current_env_name.sh"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ -f "$config_file_template" ]]; then
+    content_current_env_file="source ${config_file_template}"
+  else
+    warning "First, you need to run the script: ./setup -i"
```
Comment (rodrigosiqueira) [inline setup.sh:473]: Use sigle-quotes.

File: setup.sh [hunk 31]
```diff
@@ -446,6 +451,35 @@ function install_home()
...
+    return 2 # ENOENT
+  fi
+
+  warning 'Info:'
```
Comment (rodrigosiqueira) [inline setup.sh:477]: A warning message that prompts 'info:' does not make sense. Maybe change Info with 'Attention:'.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: setup.sh (hunk 29 above)
Comment (aquilamacedo) [reply setup.sh]: Thinking about it, I don't think it's necessary.

File: documentation/tutorials/kw-env.rst [hunk 32]
```diff
@@ -64,3 +64,32 @@ Let's say that you want to switch from one config to another, you can just use::
...
+It is especially useful when working with multiple environments within kw, as it
+helps you stay aware of the context you are working in, preventing errors and confusion.
+
+To enable this functionality, follow these steps:
+
+Run the kworkflow installation command::
+
+  ./setup -i
+
+If you have already installed kworkflow, proceed to the next step.
```
Comment (rodrigosiqueira) [inline documentation/tutorials/kw-env.rst:81]: Maybe:

If you did not install kw yet, run the below command; otherwise, go to the next step.

File: setup.sh [hunk 33]
```diff
@@ -211,6 +211,7 @@ function usage()
   say '--help      | -h     Display this usage message'
   say "--install   | -i     Install $app_name"
   say "--uninstall | -u     Uninstall $app_name"
+  say '--setup-kw-env-prompt | -s  Show current env in terminal'
```
Comment (rodrigosiqueira) [inline setup.sh:214]: Now that we are not overridden the PS1 variable, I think we don't need this. We can just install directly via ./setup -i.

File: setup.sh [hunk 34]
```diff
@@ -488,6 +488,35 @@ function install_home()
...
+  local config_file_template="${config_files_path}/kw_prompt_current_env_name.sh"
+  local shellrc="${HOME}/.bashrc"
+  local content_current_env_file
+
+  if [[ -f "$config_file_template" ]]; then
+    content_current_env_file="source ${config_file_template}"
+  else
+    warning 'First, you need to run the script: ./setup -i'
+    return 2 # ENOENT
+  fi
```
Comment (rodrigosiqueira) [inline setup.sh:512]: You'll no longer need this code if you remove the `-s`' option.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: setup.sh [hunk 35]
```diff
@@ -488,6 +488,35 @@ function install_home()
...
+
+  printf '%s\n' ''
+  warning ' -> For a better experience with kw, please, open a new terminal.'
+}
+
+function setup_bashrc_to_show_current_kw_env()
+{
+  local config_files_path="$etcdir"
+  local config_file_template="${config_files_path}/kw_prompt_current_env_name.sh"
+  local shellrc="${HOME}/.bashrc"
```
[the remark this answers is not in the document]
Comment (aquilamacedo) [reply setup.sh:504]: which one?

[the remark this answers is not in the document]
Comment (rodrigosiqueira) [reply setup.sh:500]: shellrc

**Comment by rodrigosiqueira** [comment]
I made some minor changes and merged them into the unstable branch.

Thanks!
